projects/igniteui-angular/src/lib/carousel/carousel.component.ts

```diff
@@ -315,19 +315,26 @@ export class IgxCarouselComponent implements OnDestroy {
      * @memberOf IgxCarouselComponent
      */
     public remove(slide: IgxSlideComponent) {
-        this.slides.splice(slide.index, 1);
-        this._total -= 1;
+        if (slide === this.get(slide.index)) { // check if the requested slide for delete is present in the carousel
```

Comment:
It would be nice if we add a check for slide as well.

Reply:
Done, check added.